code/obj/item.dm

```diff
@@ -882,6 +913,47 @@
 		STOP_TRACKING_CAT(TR_CAT_BURNING_ITEMS)
 	burning_last_process = src.burning
 
+/obj/item/proc/process_corrode()
+	SHOULD_NOT_SLEEP(TRUE)
+	if (src.corrosion > 0)
+		var/acid_taken_mult = 1
+		if(src.hasProperty("acidres")) //multiplier from permeability coefficient
+			acid_taken_mult = 0.01 * (100 - src.getProperty("acidres"))
+		if (istype(src.material) && src.material.hasProperty("corrosion")) //multiplier from corrosion resistance
+			acid_taken_mult = acid_taken_mult * (0.01 * (100 - src.material.getProperty("corrosion")))
+
+		var/base_melt = max(0.1 * src.corrosion,1) //calculate how much corrosion to consume, with a simple floor.
+		src.corrosion -= base_melt //consume the corrosion accordingly.
+		src.health -= base_melt * acid_taken_mult //apply the mult at this phase, so more acid resistance doesn't just prolong damage
```

Comment:
And now if I splash a security officer with a single 10u click from a beaker (or a thrown glass) filled with pacid, and they have less than 10 seconds to get to a shower before they lose their secvest *and jumpsuit* (dropping their PDA/id, belt, and pocket items on the floor). If there is a balance to be struck here, this ain't it.

Reply:
The key takeaway here imo is that pacid was balanced around only taking away mask/hat, whereas this change attempts to delete... basically all your clothes at once.

Reply:
Corresponding takeaway: Acid resilience from permeability probably won't work.